app/scripts/models/flow.js

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A model to represent a login flow, holding the data that we
+ * need to submit as part of our metrics when we perform actions
+ * during the flow.
+ *
+ * Try to fetch the flowId and flowBegin from the resume token.
+ * If unavailable there, fetch from data attributes in the DOM.
+ */
+
+define(function (require, exports, module) {
+  'use strict';
+
+  var AuthErrors = require('lib/auth-errors');
+  var Backbone = require('backbone');
+  var Cocktail = require('cocktail');
+  var ErrorUtils = require('lib/error-utils');
+  var ResumeTokenMixin = require('models/mixins/resume-token');
+  var SearchParamMixin = require('models/mixins/search-param');
```

Comment:
>  If we always mix in SearchParamMixin every time we mix in ResumeTokenMixin, is there a case for ResumeTokenMixin mixing in SearchParamMixin directly into itself?

Probably. We'd have to ensure modules that consume the ResumeTokenMixin do *not* include SearchParamMixin or else strangeness may occur because of the way Cocktail wraps functions.